extensions/wikia/AttributeServiceMWApi/AttributeServiceMWApiController.class.php

```diff
@@ -82,5 +80,44 @@ private function setErrorResponse( $message, $responseCode ) {
 		$this->response->setVal( 'message', $message );
 		$this->response->setCode( $responseCode );
 	}
+
+	private function clearUserCache( $userId ) {
+
+        // Clear user cache
+        $user = User::newFromId( $userId );
+        $user->invalidateCache();
+
+        // Clear User Profile cache
+        $userIdentityBox = new UserIdentityBox( $user );
+        $userIdentityBox->clearCache();
+    }
+
+    private function inDevEnvironment() {
+        global $wgDevelEnvironment;
+
+        return $wgDevelEnvironment === true;
+    }
+
+    /**
+     * If we're not in a development environment, clear the cache for all staging environments as well
+     * @param $userId
+     */
+    private function clearCacheInStagingEnvs( $userId ) {
+        global $wgStagingList, $wgPreviewHostname, $wgVerifyHostname;
+
+        // $wgStagingList uses the domain names "preview" and "verify" rather than the hostnames "staging-s2" and
+        // "staging-s3", so we have to manually add the those hostnames here.
+        $stagingEnvs = array_merge( $wgStagingList, [ $wgPreviewHostname, $wgVerifyHostname ] );
+        foreach ( $stagingEnvs as $stagingEnv ) {
+            $this->clearCacheInEnv( $userId, $stagingEnv );
+        }
+    }
+
+    private function clearCacheInEnv( $userId, $stagingEnv ) {
+        $wrapper = new GlobalStateWrapper( [ 'wgSharedKeyPrefix' => Wikia::getSharedKeyPrefix( $stagingEnv ) ] );
+        $wrapper->wrap( function() use ( $userId ) {
+            $this->clearUserCache( $userId );
+        } );
+    }
```

Comment:
Oops, just got a new laptop and thought I had it configured properly. Apparently not...